### ISSUE TITLE
Email invite to a new user won't remember room or space once they create account

### ISSUE THREAD

**Issue by shoqvalue** [vector-im/element-web#24793]
### Steps to reproduce

From a space or room, click Options Menu > Invite
![2023-03-10_14h39_26](https://user-images.githubusercontent.com/10272750/224412210-63eec8c5-b7a2-4300-89a6-4f8d66670bcf.png)

#### What did you expect?
That new user would a) receive invite by email, b) click the invite link, c) see the room space they were invited to, d) be prompted to create an account first, then e) verify their email and finally, f) be deposited in the space they were invited to.

#### What happened instead?
After verifying email (step e), and user is told registration was successful, there is no further mention of the room they were invited to. All they can do is "login" which puts them at a generic screen with no mention of the destination space (or room) at all.

NOTE:  To complicate this further, if the invite is for a space, and not a room, step C (which showed the room they were invited to and invited them to create an account at the bottom of screen) never appears at all.  The user has no way of knowing how to proceed.  (Michael in chat speculates that perhaps 3pid invites for spaces have been "regressed" but not sure what that meant)

![2023-03-10_14h55_00](https://user-images.githubusercontent.com/10272750/224415051-e4ba476d-f016-4deb-9e93-9d618ee6700c.png)
" 
 
![2023-03-10_14h18_12](https://user-images.githubusercontent.com/10272750/224412776-2dcb427d-3351-4f1d-8b6a-34470fbcba3e.png)

These tests were done at matrix,org with element.io to prove the probably wasn't our HS and custom element client at fault. But our setup never even got to step C above (where user is shown the space and asked to create an account). 

Something is obviously wrong or bugged, and someone who understands the invite process intimately should be reviewing both invites of new users to both rooms AND spaces.  Untangling all these permutation is way above my skill level. But I am quite confident this is NOT how it's supposed to work, and Michael seemed to confirm this, then suggested I post the issue here.   I realized it's hard to follow.  But the problem(s) are definitely there, and since no one else is complaining, it's probably a recent bug. 

### Operating system

Windows

### Browser information

110.0.5481.180 (

### URL for webapp

app.element.io

### Homeserver

matrix.org

### Will you send logs?

No

**Comment by daniellekirkwood** [comment]
Thanks so much for the feedback! We have a team who has "onboarding flow and invites" in their roadmap and this fits perfectly as a use case so we've moved this to "Triaged" and ensured that they know about it. 

👀  @pmaier1 👍

**Comment by shoqvalue** [comment]
Glad to hear it, because our fediverse-facing project is dead in the water without the ability to invite people to a space by email. We hope that road is a short one :)

**Comment by shoqvalue** [comment]
@pmaier1 How would I learn if this was fixed, so I can try again to invite folks? 
 TIA

**Comment by t3chguy** [comment]
@shoqvalue this issue would get closed if it was fixed.

**Comment by t3chguy** [comment]
I cannot reproduce this - it works fine for me. https://drive.google.com/file/d/1B9M0Q-Y8-ApA3Osk6Nmt5f9PiT0MV-dh/view?usp=sharing

**Comment by shoqvalue** [comment]
Well, we did the exact same thing and got different results (I did it 3 times). So perhaps the server was having a bad day. Let me make a video of exactly what happens here. With luck, I can't reproduce it either.  Will get to this shortly. Thanks.

**Comment by shoqvalue** [comment]
I am deeply sorry, I was asked for a video of this months ago, and I remember making it, Something failed, or I was distracted and just neglected the rather important uploading part.  I have two videos. The first is just the initial invite and signup. That always works fine. The 2nd video starts from the validation email, ,and shows that the user cannot join the room they were invited to.

Part 1 (signup process)😄
https://www.dropbox.com/scl/fi/mz9a0z14cqderajqui7x4/invitebug1of2.mp4?rlkey=e0463tppzvzptbhlejq7khq8h&dl=0

Part 2 (join room error)
https://www.dropbox.com/scl/fi/od3gncu9gi032m0kd32pe/invitebug2of2.mp4?rlkey=ttrugs2pwfz4gbyr9r6vh8gki&dl=0

**Comment by shoqvalue** [comment]
I don;t' know what you think is so different, I cannot get the same exact result as back in April where a room mentioned would appear below and wasn't exactly like a space invite, but my example above is for a room. I assumed since i never saw that again, you all changed something.  But it doesn't seem very material, since my 2nd video shows essentially the same problem I keep having, and a volunteer in the Sydent room AND an element staffer have both reproduced. I don't know what else I can provide for you.  Debugging these techs is way above my paygrade,

**Comment by t3chguy** [comment]
The issue you described here back in March was

> After verifying email (step e), and user is told registration was successful, there is no further mention of the room they were invited to. All they can do is "login" which puts them at a generic screen with no mention of the destination space (or room) at all.

Your video part 2 shows that to no longer be the case even for you, no?

**Comment by shoqvalue** [comment]
My 2nd video shows the crux of the issue, no matter what was said in March (maybe i phrased it poorly), but the problem of the user not being able to join the room is still essentially the same.   If my 2nd video doesn't make the problem clear to you, and the Sydent room volunteer you are already talking to is having essentially the sane issue, I don't know what is expected of me to convince anyone of a very obvious error.   It simply doesn't work as it's supposed to. That's all I know, except that others encountered the same problem.  I do not have any of the skills needed to understand how to give you any better information

**Comment by shoqvalue** [comment]
I have never been able to get the result in your video except that one time. And even though I could see the room I was invited to, the subsequent errors were the same.

**Comment by t3chguy** [comment]
@shoqvalue yup, as I said

> This does not mean I am discrediting the issue your video does show, but it is not the same you originally reported which no one now seems to be able to reproduce. Worth noting that I cannot reproduce your new issue on matrix.org + app.element.io which makes it seem like a misconfiguration with your Sydent or Synapse.

> It simply doesn't work as it's supposed to. 

Agreed, it definitely doesn't - in certain environments, with many pointers to it not being an Element Web/Desktop issue. If you can get it to work with other Matrix clients then that would be a counterproof to that of course.

**Comment by shoqvalue** [comment]
Oh, I think I remember now.  The test I did first was I(at your urging) using the default ELEMENT client, and that showed me the room name at the bottom (whereas our custom client and schildichat failed to even show that much_, but the subsequent errors were the same. I will do another video using the element client now (but I think others did that already).
